### ISSUE TITLE
[Feature Request] Restore from an other Dedicated server

### ISSUE THREAD

**Issue by KostasKrits** [jammsen/docker-palworld-dedicated-server#107]
### Have you read the Important information text above

- [X] Yes i did

### Describe the feature

Hello everyone.

Not sure if I am missing something here, but I came across an "issue" where my guys had rented a dedicated server and when I started playing, my DS1522+ seemed like a good host for that server. I tested and it is working fine so I have since been trying to find a way to move over the server data.

I tried exporting container content and settings from the Synology container manager to look into the export and figure out if I can replace some files and then import it back as a new container but I haven't found anything yet.

Moving the save data from the online dedicated server to my windows steam dedicated server worked fine though since the folder structure was pretty much the same.

So in conclusion, it would be nice if such a feature existed, or if it's doable, someone could tell me what I am missing here!

### Additional information

- [ ] Would you be willing to help implement this feature?

### Final checks

- [X] I understand that this is a feature request function. I also unterstand that if i use this to report a bug report, that my issue might be un-answered closed.

**Comment by jammsen** [comment]
Hey @KostasKrits are you asking if i would add a migration feature to a docker-image for hosting Palworld?
Not sure i understand you, tbh.

**Comment by Deglingoss** [comment]
Hello there,

ty vm for your hard wrok. 

We are triyng to migrate data form a windows local server into your docker with no success. Save data structure seems the same.
We even changed DedicateServerName to match the origin server ID. But still not able to load existing profiles when connecting.

We also miss something. And we don't find such a simple doc on how to migrate data from one server to another.

If you could point us to the right direction that would be great.

Regards

**Comment by KostasKrits** [comment]
> Hey @KostasKrits are you asking if i would add a migration feature to a docker-image for hosting Palworld? Not sure i understand you, tbh.

I am not specifically asking if you are planning to add this feature. I am just asking if you've done it at some point, or if it's even possible to migrate save data from a steam windows server to the docker server.

**Comment by KostasKrits** [comment]
> Hello there,
> 
> ty vm for your hard wrok.
> 
> We are triyng to migrate data form a windows local server into your docker with no success. Save data structure seems the same. We even changed DedicateServerName to match the origin server ID. But still not able to load existing profiles when connecting.
> 
> We also miss something. And we don't find such a simple doc on how to migrate data from one server to another.
> 
> If you could point us to the right direction that would be great.
> 
> Regards

Have you done it on a Synology NAS by any chance? I have exported the container and I cannot seem to find the save file. But I am also not sure if I am looking in the right location

**Comment by Deglingoss** [comment]
> > Hello there,
> > ty vm for your hard wrok.
> > We are triyng to migrate data form a windows local server into your docker with no success. Save data structure seems the same. We even changed DedicateServerName to match the origin server ID. But still not able to load existing profiles when connecting.
> > We also miss something. And we don't find such a simple doc on how to migrate data from one server to another.
> > If you could point us to the right direction that would be great.
> > Regards
> 
> Have you done it on a Synology NAS by any chance? I have exported the container and I cannot seem to find the save file. But I am also not sure if I am looking in the right location

We deployed the docker in a Synology. We access it's contents connecting to the SYno through ssh and becoming root and then executing:

docker exec -it [container_name] /bin/bash

Saved folder fo the game server are in /palworld/Pal/Saved. Those are the runtime files.

Jammsen developped a backup script that saves this content in :

/palworld/backups

Regards

**Comment by KostasKrits** [comment]
> > > Hello there,
> > > ty vm for your hard wrok.
> > > We are triyng to migrate data form a windows local server into your docker with no success. Save data structure seems the same. We even changed DedicateServerName to match the origin server ID. But still not able to load existing profiles when connecting.
> > > We also miss something. And we don't find such a simple doc on how to migrate data from one server to another.
> > > If you could point us to the right direction that would be great.
> > > Regards
> > 
> > 
> > Have you done it on a Synology NAS by any chance? I have exported the container and I cannot seem to find the save file. But I am also not sure if I am looking in the right location
> 
> We deployed the docker in a Synology. We access it's contents connecting to the SYno through ssh and becoming root and then executing:
> 
> docker exec -it [container_name] /bin/bash
> 
> Saved folder fo the game server are in /palworld/Pal/Saved. Those are the runtime files.
> 
> Jammsen developped a backup script that saves this content in :
> 
> /palworld/backups
> 
> Regards

Okay I see. Seems like I had created a folder for the container which seems like I didnt have too. I had named it palworld so somehow it overlapped with the original palworld folder where the data are saved.

Now, maybe a newbie question, how do I move the save files from my computer to the container itself via ssh?

**Comment by jammsen** [comment]
> Hello there,
> 
> ty vm for your hard wrok.
> 
> We are triyng to migrate data form a windows local server into your docker with no success. Save data structure seems the same. We even changed DedicateServerName to match the origin server ID. But still not able to load existing profiles when connecting.
> 
> We also miss something. And we don't find such a simple doc on how to migrate data from one server to another.
> 
> If you could point us to the right direction that would be great.
> 
> Regards

Have you looked at this?
https://docs.google.com/document/d/1uzyT-02JqPz2KWs4rMEM3tzoEVwoKDTInroDIF0Y7s0/edit#heading=h.4nl2uzwnpzz2

That document seems to be from Pocketpair, in the Discord its linked a few times, maybe it helps?

**Comment by jammsen** [comment]
> > Hey @KostasKrits are you asking if i would add a migration feature to a docker-image for hosting Palworld? Not sure i understand you, tbh.
> 
> I am not specifically asking if you are planning to add this feature. I am just asking if you've done it at some point, or if it's even possible to migrate save data from a steam windows server to the docker server.

Not done, no experience, try the Pocketpair Discord maybe?

**Comment by Deglingoss** [comment]
> Okay I see. Seems like I had created a folder for the container which seems like I didnt have too. I had named it palworld so somehow it overlapped with the original palworld folder where the data are saved.
> 
> Now, maybe a newbie question, how do I move the save files from my computer to the container itself via ssh?

Hi again, 
In your NAS you have to active the SSH , then you open Windows CMD

ssh yourNASadminuser@yourNASip
yourNASpassword

sudo -i
yourNASpassword

access to your container
docker exec -it containername /bin/bash

after you can write : 
cd /palworld/Pal/Saved/

My friends helped me but now I'm alone so can't help you anymore for the moment. 
Maybe this can help people to have more information about savegames : 
https://github.com/xNul/palworld-host-save-fix

Regards

**Comment by KostasKrits** [comment]
> > Okay I see. Seems like I had created a folder for the container which seems like I didnt have too. I had named it palworld so somehow it overlapped with the original palworld folder where the data are saved.
> > Now, maybe a newbie question, how do I move the save files from my computer to the container itself via ssh?
> 
> Hi again, In your NAS you have to active the SSH , then you open Windows CMD
> 
> ssh yourNASadminuser@yourNASip yourNASpassword
> 
> sudo -i yourNASpassword
> 
> access to your container docker exec -it containername /bin/bash
> 
> after you can write : cd /palworld/Pal/Saved/
> 
> My friends helped me but now I'm alone so can't help you anymore for the moment. Maybe this can help people to have more information about savegames : https://github.com/xNul/palworld-host-save-fix
> 
> Regards

Yeah I have been that far. Need to see how I can actually move the files now

**Comment by tipanic** [comment]
If you want to transfer files from Windows to a Linux host, use WinSCP, but you need to change the Owner and maybe the rights after you imported them. But only If your docker environment runs with another user than root. 

I migrated my server from a different Palworld docker to this one yesterday. The only thing i did not, was to create every Character that was previously on my old server. I just started the server, tested the login, stopped docker, imported my folders from the old server backup and changed the owner to my docker user. All users could login successfully and no one had any loss in progress.

**Comment by jammsen** [comment]
Im closing this issue as resolved, feel free to try it out and reopen again if needed.

If you like this project, please consider giving this repo and the docker-hub-repo a Star.

**Comment by YuZongYangHi** [comment]
> 如果您想将文件从 Windows 传输到 Linux 主机，请使用 WinSCP，但您需要更改所有者，也许还需要更改导入后的权限。但前提是您的 docker 环境使用除 root 之外的其他用户运行。
> 
> 昨天我将我的服务器从另一台 Palworld docker 迁移到了这个。我唯一没有做的就是创建以前在我的旧服务器上的每个角色。我刚刚启动服务器，测试登录，停止 docker，从旧服务器备份导入我的文件夹，并将所有者更改为我的 docker 用户。所有用户均能成功登录，没有任何人出现任何进度损失。

Can you elaborate on that? My current conversion from windows to linux failed
